File: ruby/leap/leap.rb
module BookKeeping
  VERSION = 2
end

class Year
  def self.leap?(year)
    leap_year = false
    leap_year = true if year % 4 == 0
    leap_year = false if year % 100 == 0
    leap_year = true if year % 400 == 0
    leap_year
  end
end

File: go/leap/leap.go
// Leap stub file

package leap

// testVersion should match the targetTestVersion in the test file.
const testVersion = 2

func IsLeapYear(year int) bool {
	switch {
	case year%400 == 0:
		return true
	case year%100 == 0:
		return false
	case year%4 == 0:
		return true
	default:
		return false
	}
}
